### DISCUSSION TITLE
Max size on JSON response

### DISCUSSION THREAD

**Discussion by setroc** [cesanta/mongoose#2255] [Ask questions, get answers] [answered]
Hello, I recently made a Rest API, where in a specific route I send a JSON as a response. I would like to know if there is any limit to the size of the JSON that can be sent as a response by the server? is it possible to configure this?

<img width="531" alt="image" src="https://github.com/cesanta/mongoose/assets/38026419/5a43f109-d0cb-4886-abcb-719265c094a2">

**Comment by scaprile** [comment, marked as answer]
Well... I fail to see the "JSON" specifics in your question, you are just calling https://mongoose.ws/documentation/#mg_http_reply
Mongoose uses dynamic memory allocation. Your readJSON(), whatever it does, will return a `const char *`. `mg_http_reply()` will grab that string and try to put it on Mongoose buffer to be sent to TCP, The buffer will try to resize to fit the contents, and if it can't, you'll see an out of memory message in the log
https://github.com/cesanta/mongoose/blob/d013e0fc78db3ecc1383477ee8c6c216219fcd6a/src/http.c#L336-L352
https://github.com/cesanta/mongoose/blob/d013e0fc78db3ecc1383477ee8c6c216219fcd6a/src/printf.c#L26-L43
https://github.com/cesanta/mongoose/blob/d013e0fc78db3ecc1383477ee8c6c216219fcd6a/src/iobuf.c#L17-L42
